src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.CitizenDao;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class<?> clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor<?> constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(BetDao.class)) {
+                field.setAccessible(true);
+                field.set(instance, Factory.getBetDao());
+            }
+            if (field.getAnnotation(Inject.class) != null
+                    && field.getType().equals(CitizenDao.class)) {
+                field.setAccessible(true);
+                field.set(instance, Factory.citizenDao());
```

Comment:
Let's get rid of repeating code in here. We should check the Inject annotation presence first, then make the fields accessible and then only we should check the class equality and set field's value.